### DISCUSSION TITLE
a training bug

### DISCUSSION THREAD

**Discussion by rechawine** [pykt-team/pykt-toolkit#59] [Q&A] [answered]
when i use sakt_model to trian, there is a error:

`  File "/home/data/lzq22/aaai23/pykt-toolkit/examples/wandb_train.py", line 77, in main
    train_loader, valid_loader = init_dataset4train(dataset_name, model_name, data_config, fold, batch_size)
ValueError: too many values to unpack (expected 2)`

what should i do to solve this problem?

**Comment by meta-tabchen** [comment, marked as answer]
Hi, can you provide more details, such as what command and dataset you running? 

I tested assist2009 and peiyou(we published in [AAAI2023 workshop](https://codalab.lisn.upsaclay.fr/competitions/8087)) everything is correct. The following are my codes:

```shell
python wandb_sakt_train.py --use_wandb=0 --dataset_name=assist2009
python wandb_sakt_train.py --use_wandb=0 --dataset_name=peiyou
```

And you can try to pull the lasted codes, and run the commands I provided, hope that will helps you.

**Reply by rechawine** [reply]
thanks, the problem is solved
but there is another problem, i train the qdkt model correctly, and try to inference, it has some error:

`Traceback (most recent call last):
  File "wandb_eval.py", line 96, in <module>
    main(params)
  File "wandb_eval.py", line 59, in main
    dfinal = model.evaluate_multi_ahead(data_config,batch_size=32,ob_portions=ratio,accumulative=use_pred)
  File "../pykt/models/que_base_model.py", line 481, in evaluate_multi_ahead
    auc,acc = self._evaluate_multi_ahead_help(data_config,batch_size=batch_size,ob_portions=ob_portions,acc_threshold=acc_threshold)
  File "../pykt/models/que_base_model.py", line 448, in _evaluate_multi_ahead_help
    y = self.predict_one_step(data,process=False)[:,-1].detach().cpu().numpy().flatten()
  File "../pykt/models/qdkt.py", line 67, in predict_one_step
    outputs = self.model(data_new['cq'].long(),data_new['cc'],data_new['cr'].long(),data=data_new)
  File "/home/user/miniconda3/envs/pykt/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "../pykt/models/qdkt.py", line 41, in forward
    y = (y * F.one_hot(data['qshft'].long(), self.num_q)).sum(-1)
RuntimeError: CUDA error: device-side assert triggered`

**Reply by meta-tabchen** [reply]
Hi, thank for you your attention.
We run the dataset with qDKT model, and the training and eval are correct. Would you please provide more details? And the qDKT and IEKT model is not support testing with the file pykt_test.csv now, you can try other models.
